### PR TITLE
bpo-15108: Prevent accessing the result tuple from Python in PySequence_Tuple

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue15108

### CODE REVIEW

File: Objects/abstract.c [hunk 1]
```diff
@@ -1993,6 +1993,9 @@ PySequence_Tuple(PyObject *v)
     if (result == NULL)
         goto Fail;
 
+    // bpo-15108: Code can access the result tuple while being
+    // incomplete when calling PyIter_Next().
+    PyObject_GC_UnTrack(result);
```
Comment (vstinner) [inline Objects/abstract.c:1998]: I would prefer to add a new API to create an untracked tuple: https://bugs.python.org/issue15108#msg387029

Comment (pablogsal) [reply Objects/abstract.c:1998]: Let's discuss int the issue, then I can modify the PR :)

**Comment by rhettinger** [comment]
FWIW:  Traditionally any "crasher" that depended on *gc.get_referrers* was not considered a bug.  These do not seem to affect "normal code" and are hard to fix without doing brain surgery to Python.

**Comment by pablogsal** [comment]
Yeah, I have to say that I am not very convinced about this change because it also fixes a bigger problem only on one place.

What worries me the most here is not using `gc.get_referrers`, but the GC crashing because it does a GC pass when the tuple is invalid.

**Pull request closed without merging**